packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -163,12 +198,23 @@ export function useLocationDetailView(
       dispatchStoreAction({ type: 'RESET_LOCATION' });
 
       handleList({
+        config: getConfig(),
         // @todo: prefix should not be required to refresh
         prefix: current?.prefix ?? '',
         options: { reset: true },
       });
       dispatchStoreAction({ type: 'RESET_ACTION_TYPE' });
       dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
     },
+    onSearch: (query, includeSubfolders) => {
+      if (hasInvalidPrefix) return;
+      const searchOptions = {
+        ...listOptions,
+        delimiter: includeSubfolders ? undefined : listOptions.delimiter,
+        search: { query, filterKey: 'key' as const },
+      };
+      handleReset();
+      handleList({ config: getConfig(), prefix, options: searchOptions });
+    },
```

Comment:
We probably need to reset selection state as well in case they selected files we're going to replace.

```suggestion
      dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
    },
```

Not sure if ` dispatchStoreAction({ type: 'RESET_ACTION_TYPE' });` might also be necessary?

Reply:
Good call, i think we can treat it like a refresh, so `dispatchStoreAction({ type: 'RESET_ACTION_TYPE' });` might not be necessary unless we want to do that `onRefresh` as well.